spec/unit/plugins/windows/filesystem_spec.rb

```diff
@@ -70,9 +70,99 @@
     allow(plugin).to receive(:collect_os).and_return(:windows)
   end
 
+  describe "the plugin" do
+    context "when there are no volume names" do
+      before do
+        allow(plugin).to receive(:logical_info).and_return(plugin.logical_properties(logical_disks_instances))
+        allow(plugin).to receive(:encryptable_info).and_return(plugin.encryption_properties(encryptable_volume_instances))
+        plugin.run
+      end
+
+      it "returns space information" do
+        puts plugin[:filesystem]
+        {
+          "kb_size" => 10000,
+          "kb_available" => 100,
+          "kb_used" => 9900,
+          "percent_used" => 99,
+        }.each do |k, v|
+          expect(plugin[:filesystem]["C:"][k]).to eq(v)
+          expect(plugin[:filesystem]["D:"][k]).to eq(v)
+          expect(plugin[:filesystem2]["by_pair"][",C:"][k]).to eq(v)
+          expect(plugin[:filesystem2]["by_pair"][",D:"][k]).to eq(v)
+        end
+      end
+
+      it "returns disk information" do
+        {
+          "fs_type" => "ntfs",
+          "volume_name" => "",
+          "encryption_status" => "FullyDecrypted",
+        }.each do |k, v|
+          expect(plugin[:filesystem]["C:"][k]).to eq(v)
+          expect(plugin[:filesystem2]["by_pair"][",C:"][k]).to eq(v)
+        end
+
+        {
+          "fs_type" => "fat32",
+          "volume_name" => "",
+          "encryption_status" => "EncryptionInProgress",
+        }.each do |k, v|
+          expect(plugin[:filesystem]["D:"][k]).to eq(v)
+          expect(plugin[:filesystem2]["by_pair"][",D:"][k]).to eq(v)
+        end
+      end
+    end
+
+    context "when there are volume names" do
+      before do
+        ldi = logical_disks_instances
+        ldi.each_with_index { |d, i| d["volume_name"] = "Volume #{i}" }
+        allow(plugin).to receive(:logical_info).and_return(plugin.logical_properties(ldi))
+        allow(plugin).to receive(:encryptable_info).and_return(plugin.encryption_properties(encryptable_volume_instances))
+        plugin.run
+      end
+
+      it "returns space information" do
+        puts plugin[:filesystem2]["by_pair"]
```

Comment:
We could remove this, and another one at L#82, the rest LGTM ✔️

Reply:
Ooops!! My debugging got left in there, sorry. Nice catch.